pkgs/development/python-modules/tensorflow/default.nix

```diff
@@ -540,18 +573,19 @@ in buildPythonPackage {
     hello = tf.constant("Hello, world!")
     tf.print(hello)
 
+    # need to move keras-based test to keras package
```

Comment:
yeah let's just replace with a pure TF test here to make our lives easier

Reply:
I moved it here: https://github.com/NixOS/nixpkgs/pull/256470

Open to suggestions for a TF-only test.

Reply:
<img width="837" alt="Screenshot 2023-09-21 at 1 41 03 PM" src="https://github.com/NixOS/nixpkgs/assets/226872/0cf5e2fe-c568-48d6-b8f6-4ea09c922f71">

on second thought, the test runs perfectly fine on TF 2.13.0 without modification

Reply:
Yes, I know, many features still work with `keras` 2.11.0 and `tensorflow` 2.13.0. But keeping the test means we need to depend on `keras`, which I consider to be wrong. The 2.13.1 `keras` update (https://github.com/NixOS/nixpkgs/pull/256470) was only possible by depending on `tensorflow` since `fetchPypi` could not grab it, and I had to resort to building from source. And more generally, as long as we block `keras` source builds by requiring it to be present before `tensorflow`, we will always be bound to use the upstream build artifacts in `keras`, instead of being able to independently build from source. Which is why I consider the dependency to be a bug (https://github.com/NixOS/nixpkgs/issues/255733).

No problem, though. I can come up with a small TF-only test script. Maybe some `autodiff` example using `GradientTape`.

Reply:
Yeah, here's an example we could use:
```py
x = tf.Variable(3.0)
with tf.GradientTape() as tape:
    y = x * x
dy_dx = tape.gradient(y, x).numpy()
assert dy_dx == 6.0
```
I'm not too picky on the specifics, but it'd be nice to replace the existing test as opposed to dropping it

Reply:
I agree. I committed a test that uses some more tf functions.